### ISSUE TITLE
Bad inference for destructured object parameter with default value

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#22644]
**TypeScript Version:**  2.7.2

**Search Terms:** object literal parameter inference destructuring default value

**Code**

```ts
const fn1 = (options: { headers?: {} }) => {};
// No error as expected
fn1({ headers: { foo: 1 } });

const fn2 = ({ headers = {} }) => {};
// Unexpected error: Property 'foo' does not exist on type '{}'.
fn2({ headers: { foo: 1 } });
```

**Comment by RyanCavanaugh** [comment]
Regression from 2.6 -> 2.7

**Comment by ahejlsberg** [comment]
This is an issue exposed by #19513. We're not properly widening types in types implied by object destructuring patterns. Looks like a fairly simple fix.

**Comment by OliverJAsh** [comment]
Thank you!
